simdpp/algorithm/all_of.h

```diff
@@ -0,0 +1,69 @@
+/*  Copyright (C) 2018  Povilas Kanapickas <povilas@radix.lt>
+
+    Distributed under the Boost Software License, Version 1.0.
+    (See accompanying file LICENSE_1_0.txt or copy at
+    http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+#ifndef LIBSIMDPP_SIMDPP_ALGORITHM_ALL_OF_H
+#define LIBSIMDPP_SIMDPP_ALGORITHM_ALL_OF_H
+
+#ifndef LIBSIMDPP_SIMD_H
+#error "This file must be included through simd.h"
+#endif
+
+#include <algorithm>
+#include <simdpp/detail/insn/load.h>
+#include <simdpp/algorithm/helper_input_range.h>
+
+namespace simdpp {
+   namespace SIMDPP_ARCH_NAMESPACE {
+
+      template<typename T, typename UnaryPredicate>
+         bool all_of(T const* first, T const* last, UnaryPredicate pred)
+      {
+#ifndef NDEBUG //precondition debug mode           
+         if (!first)
+            throw std::runtime_error("all_of - null ptr first.");
+         if (!last)
+            throw std::runtime_error("all_of - null ptr last.");
+#endif
+
+         using simd_type_T = typename typetraits<T>::simd_type;
+         using simd_mask_T = typename typetraits<T>::simd_mask_type;
+
+         //define loopcounter
+         const auto simd_size = simd_type_T::base_length;
+
+         //note enforce that input is aligned when we start the main simd loop
+         const auto range = helper_input_range(first, last);
+         const auto size_prologue_loop = range.first;
+         const auto size_simd_loop = range.second;
+
+         //prologue
+         auto lastprologue = first + size_prologue_loop;
+         if(!std::all_of(first, lastprologue, pred)) return false;
```

Comment:
May i miss something. But epilogue and prologue must be use in  case where input are not a multiple of size of simd registers.

Prologue
Think about input vector of 7 unsigned int, we will never use mainsimdloop nor epilogue part. But we unsure that we could transparently use simdpp::all_of instead of std::all_of

In a same manner, if the input size is 33 aligned unsigned int, we don't use epilogue at all, in SSE2 case, 32 element pass through main simd loop and the remaining element use std versions